openerp/addons/base/ir/ir_attachment.py

```diff
@@ -431,6 +433,9 @@ def write(self, cr, uid, ids, vals, context=None):
         # remove computed field depending of datas
         for field in ['file_size', 'checksum']:
             vals.pop(field, False)
+        # if mimetype not given, compute it !
+        if 'mimetype' not in vals:
+            vals['mimetype'] = self._compute_mimetype(vals)
```

Comment:
Problem is when we created attachment for pdf file. after that i want to change that attachment and replace doc file with pdf file. but mimetype is not updated it's take pdf format.
because I have add condition in write method.